src/osd/scrubber/pg_scrubber.cc

```diff
@@ -2776,24 +2777,38 @@ ReplicaReservations::no_reply_t::no_reply_t(
   ReplicaReservations& parent,
   std::string_view log_prfx)
     : m_osds{osds}
-    , m_conf{conf}
     , m_parent{parent}
     , m_log_prfx{log_prfx}
 {
-  using namespace std::chrono;
   auto now_is = clock::now();
-  auto timeout =
-    conf.get_val<std::chrono::milliseconds>("osd_scrub_reservation_timeout");
-
-  m_abort_callback = new LambdaContext([this, now_is]([[maybe_unused]] int r) {
-    // behave as if a REJECT was received
-    m_osds->clog->warn() << fmt::format(
-      "{} timeout on replica reservations (since {})", m_log_prfx, now_is);
-    m_parent.handle_no_reply_timeout();
-  });
+  auto timeout = conf.get_val<milliseconds>("osd_scrub_reservation_timeout");
+
+  m_abort_callback = new LambdaContext(
+      [this, now_is, pgid = m_parent.m_pg->pg_id]([[maybe_unused]] int r) {
+	// behave as if a REJECT was received
+	if (!m_abort_callback) {
+	  // already cancelled
+	  return;
+	}
+	m_abort_callback = nullptr;
```

Comment:
This is a modification of the PGScrubber structure prior to holding the PG lock.